### ISSUE TITLE
Molten core

### ISSUE THREAD

**Issue by Coldsteel12** [Atlantiss/BugTracker#10071]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

[//]: # (On the Bugtracker all titles of reports must be written in English so that everyone, no matter which language they use everyday, could search and check if the problem already has been reported by another player. When describing the problem you can use English as well as Polish and in nearest future - others. It is important to determine at the beginning of a title in which language it is written by using symbols [EN] or [PL].)

[//]: # (Enclose links to things related to the bug using http://wowhead.com or http://cata.openwow.com)
[//]: # (Write your tickets according to the format:)
[//]: # ([EN][Quest][Zone] A Vision of the Past)
[//]: # ([EN][NPC] Thoralius the Wise)
[//]: # ([EN][Spell][Class] Frostfire Bolt)
[//]: # ([PL][Talent][Class] Brain Freeze)
[//]: # ([EN][Glyph][Class] Glyph of Frostfire Bolt)
[//]: # ([EN][Npc][Drop] Thoralius the Wise)
[//]: # ([EN][Web] Armory doesnt work)

**Description**: Demonology warlock's talent "Molten Core" Is bugged. 
**Current behaviour**: If warlock get this buff from talent, while Insinirate is casting, exactly this insinirate doesnt improve for 18% (damage buff), but lock losts one charge of this proc (3 charges). I mean: i am casting insi, i got a proc, when cast ends, i lose 1 charge, this insi has the same damage. It's wrong.

**Expected behaviour**: Acoording Cata system of procs, spell should be modified according you buffs, which you have, when you just start casting. So, i think, warlock should not lost 1 proc, when this happens. He just need to cast 3 improved insinirate after that cast.

**Steps to reproduce the problem**:

1. Just check it yourself
2. Fix
3. Release

**Revision**: rev 5790

**Comment by Aloise3** [comment]
The video shows the buff staying if Incinerate is already being cast. (4:35-4:42)
https://youtu.be/93iWoo3JLbs

**Comment by Coldsteel12** [comment]
Yes,on this video for example at 4:40 you can see how it should be. Look on the cast timer. Lock casts incinirate for 1.8 sec ,At this time he got Molten Core, but next 3 incis were 1.3 sec.

**Comment by Coldsteel12** [comment]
do we have any chance to see fixed molten core?
